src/NoteEditPage.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import ReactRouterPropTypes from 'react-router-prop-types';
+
+import { NoteEditPage } from '@folio/stripes/smart-components';
+
+import { retrieveNoteReferredEntityDataFromLocationState } from './util';
+
+export default class NoteEditRoute extends Component {
+  static propTypes = {
+    history: ReactRouterPropTypes.history.isRequired,
+    location: ReactRouterPropTypes.location.isRequired,
+    match: PropTypes.shape({
+      params: PropTypes.shape({
+        id: PropTypes.string.isRequired,
+      }).isRequired,
+    }).isRequired,
+  };
+
+  goToNoteView = () => {
+    const {
+      match,
+      history,
+      location,
+    } = this.props;
+
+    const { id } = match.params;
+    const noteViewUrl = `/users/notes/${id}`;
+
+    history.replace({
+      pathname: noteViewUrl,
+      state: location.state,
+    });
+  }
+
+  render() {
+    const {
+      match,
+      location: { state },
+    } = this.props;
+
+    const noteId = match.params.id;
+    const referredEntityData = retrieveNoteReferredEntityDataFromLocationState(state);
```

Comment:
@MaksymDryha it looks like every time you want to use `retrieveNoteReferredEntityDataFromLocationState` you have to first extract state from the location object. Perhaps it would be cleaner to do that inside `retrieveNoteReferredEntityDataFromLocationState` and just pass the `location` as a param to `retrieveNoteReferredEntityDataFromLocationState`?

Reply:
Hmm maybe it's not a big deal since you still need to extract location from props anyway.

Reply:
@mkuklis, I think it's better to pass just `state`, because if we pass the whole `location` object, then `retrieveNoteReferredEntityDataFromLocationState` will receive a lot of things which it doesn't want to care about